plugins/env/app/models/environment_variable.rb

```diff
@@ -19,11 +19,12 @@ class << self
     # but return a value with a helpful message
     # also used by an external plugin
     def env(project, deploy_group, preview: false, resolve_secrets: true)
-      variables = nested_variables(project)
-      variables.sort_by!(&:priority)
-      env = variables.each_with_object({}) do |ev, all|
-        all[ev.name] = ev.value if !all[ev.name] && ev.matches_scope?(deploy_group)
-      end
+      env =
+        if (env_repo_name = ENV["DEPLOYMENT_ENV_REPO"]) && project.use_env_repo
+          env_vars_from_repo(env_repo_name, project, deploy_group)
+        else
+          env_vars_from_db(project, deploy_group)
```

Comment:
the db env var UI is still visible, so that would be kinda confusing for users ?
maybe merge them ?

Reply:
? I think when the use_env_repo is true then the db vars are not displayed

Reply:
they are still there ... no if/else logic, but just `if` that adds a new field
so either hide them from the UI or merge them ... merge would allow partial migrations, but if/else would mean a cleaner separation 🤷‍♂️